src/pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -283,6 +283,24 @@ def _check_config_and_set_final_flags(
                 f" Available names are: {', '.join(self._accelerator_types)}."
             )
 
+        # mps accelerator incompatible with ddp-family
+        ddp_classes = (
+            DDPStrategy,
+            DDPSpawnShardedStrategy,
+            DDPShardedStrategy,
+            DDPFullyShardedNativeStrategy,
+            DDPFullyShardedStrategy,
+            DDPSpawnStrategy,
+        )
+        is_ddp_str = isinstance(strategy, str) and "ddp" in strategy
+        if (
+            accelerator is not None
+            and strategy is not None
+            and (isinstance(accelerator, str) and accelerator == "mps")
+            and (is_ddp_str or isinstance(strategy, ddp_classes))
```

Comment:
```suggestion
            and (is_ddp_str or isinstance(strategy, ParallelStrategy))
```
this should cover all kinds of parallelism